### DISCUSSION TITLE
Request for Support of Document Relationships between Indexes

### DISCUSSION THREAD

**Discussion by Karman40** [meilisearch/product#605] [Feedback & Feature Proposal]
Dear MeiliSearch Team,

I hope this message finds you well. I have been using MeiliSearch for some time now, and I am impressed with its speed and accuracy in handling full-text search queries.

However, I have a feature request that I believe would greatly enhance the capabilities of MeiliSearch. I would like to request that the future versions of MeiliSearch support document relationships between indexes. Currently, it is not possible to manage relationships between documents stored in different indexes, which can limit the use cases for MeiliSearch.

Adding support for document relationships between indexes would make MeiliSearch more versatile and allow it to be used in a wider range of applications. I believe this feature would be valuable for many MeiliSearch users and would help to further differentiate MeiliSearch from other search engines.

Thank you for your time and consideration of this request. I look forward to your response and the potential for this feature in future releases of MeiliSearch.

Best regards,
Zsombor KÁRMÁN

**Comment by macraig** [comment]
Hi @Karman40 , apologies for the delayed reply 🙇 Thank you for taking the time to leave your feedback, I'm happy to hear that you find Meilisearch fast and accurate! 🚀

Could you expand on your use case for supporting document relationships between indexes? How would this help you? How is it limiting you currently? I'd love to better understand your needs to document them and add them to our backlog!

**Reply by Karman40** [reply]
Hi @macraig.

I am using MongoDB as my primary database, with two example collections:

- servers
- organizations

The schema for the "servers" collection is as follows: `id: ..., org: org.id, etc...`. When searching for a server using MeiliSearch, I am unable to filter the results based on the organization data, only org.id.
Combining the organization and server information into a single document within MeiliSearch would require a significant amount of work for both the initial setup and managing updates whenever changes occur.


Or, for example, managing relationships between users and profiles to make them searchable.
![image](https://user-images.githubusercontent.com/32225709/237058838-ddd2d21e-db44-4a7b-b6a9-f41e7ad0f145.png)

**Reply by macraig** [reply]
Thanks for sharing your use case @Karman40. I'm going to leave this discussion open so that others who might be experiencing similar challenges can contribute.

However, we want to be transparent about our current priorities. Our team is presently occupied with other pressing commitments and projects, so we may not be able to address this specific request right away. That being said, we genuinely appreciate your feedback as we're always looking for ways to improve our offerings.

I hope you understand our position and we'll certainly keep your idea in mind for the future. Please feel free to continue to provide input and share your thoughts - your input is extremely valuable to us!

**Reply by Karman40** [reply]
@macraig what do you think, could you support this in the near future?

**Reply by macraig** [reply]
@Karman40 this isn't on our near-term roadmap at the moment, but I'll make sure to update the issue if we prioritize it.

**Comment by ccbiozhaw** [comment]
hello, I also think the ability to expand through relationships would be very useful. I have very dynamic data with lots of relations and it is a headache to keep it all in sync nicely.

**Reply by macraig** [reply]
Thank you for sharing your feedback @ccbiozhaw !
